src/nvim/api/win_config.c

```diff
@@ -185,9 +185,7 @@ Window nvim_open_win(Buffer buffer, Boolean enter, Dict(float_config) *config, E
   if (!wp) {
     return 0;
   }
-  if (enter) {
-    win_enter(wp, false);
-  }
+  new_win_ext(wp, enter, false, buf);
```

Comment:
This doesn't save any code, it only adds redirection. My general question is whether some part of the `win_enter` codepath should already trigger WinNew. That requires looking at existing callers to `win_enter`